### PR TITLE
ARROW-53 Add Support for Boolean Type

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: bindings/python/docs/source/supported_types.rst [hunk 1]
```diff
@@ -27,6 +27,8 @@ Support for additional types will be added in subsequent releases.
      - :class:`~py.int`, :class:`bson.int64.Int64`, an instance of :meth:`pyarrow.int64`
    * - UTC datetime
      - an instance of :class:`~pyarrow.timestamp` with ``ms`` resolution, :class:`py.datetime.datetime`
+   * - Boolean
+     - an instance of :class:`~pyarrow.bool_`, :class:`~py.bool`
```
Comment (blink1073) [inline bindings/python/docs/source/supported_types.rst:31]: I've been putting these in the same order as their [type number](https://docs.mongodb.com/manual/reference/bson-types/)

**Review by blink1073 — CHANGES_REQUESTED** [review]
Nice work!  Just one suggestion about the docs

File: bindings/python/docs/source/supported_types.rst (hunk 1 above)
Comment (juliusgeo) [reply bindings/python/docs/source/supported_types.rst]: Ah ok, makes sense. Done.

**Review by blink1073 — APPROVED** [review]
Thanks!

File: bindings/python/test/test_bson.py [hunk 2]
```diff
@@ -154,3 +154,24 @@ def test_object_id_type(self):
         arr = pa.ExtensionArray.from_storage(ObjectIdType(), storage_array)
         result = self.serialize_array(arr)
         assert result.type._type_marker == ObjectIdType._type_marker
+
+
+class TestBooleanType(TestBsonToArrowConversionBase):
+    def setUp(self):
+        self.schema = Schema({'_id': ObjectId,
```
Comment (ShaneHarvey) [inline bindings/python/test/test_bson.py:161]: Can we remove the `_id` from this test to make is smaller and easier to read? IIUC _id+ObjectId serves no purpose in this test.

File: bindings/python/test/test_bson.py [hunk 3]
```diff
@@ -154,3 +154,24 @@ def test_object_id_type(self):
...
+                              'data': bool_()})
+        self.context = PyMongoArrowContext.from_schema(
+            self.schema)
+
+    def test_simple(self):
+        ids = [ObjectId() for i in range(4)]
+        docs = [{'_id': ids[0], 'data': True},
+                {'_id': ids[1], 'data': False},
+                {'_id': ids[2], 'data': True},
+                {'_id': ids[3], 'data': False}]
```
Comment (ShaneHarvey) [inline bindings/python/test/test_bson.py:171]: Please test missing and non-bool 'data' fields as well to test the append_null behavior of `process_bson_stream`.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/test/test_bson.py (hunk 2 above)
Comment (juliusgeo) [reply bindings/python/test/test_bson.py:161]: Done.

File: bindings/python/test/test_bson.py (hunk 3 above)
Comment (juliusgeo) [reply bindings/python/test/test_bson.py:171]: Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM!

**Pull request merged**